tensorboard/plugins/graph/tf_graph_dashboard/tf-graph-dashboard.html

```diff
@@ -193,6 +199,7 @@
       value: true,  // TODO(@chihuahua): Reconcile this setting with Health Pills.
       readOnly: true,
     },
+    _selectedDataset: Number,
```

Comment:
This doesn't look like it responds to changes in the hash. Other plugins have bidirectional linkage.

Is there a reason that you don't want to use
```ts
properties: {
  _selectedDatasetName: {
    type: String,
    notify: true,
    value: storage.getStringInitializer('run', /*default=*/'', /*useLocalStorage=*/false),
    observer: '_selectedDatasetNameObserver',
  },
},
_selectedDatasetNameObserver: storage.getStringObserver(
  'run', /*default=*/'', /*useLocalStorage=*/false),
```
that then has an additional observer on `_selectedDatasetName` that looks up the index and sets `_selectedDataset` accordingly?

Reply:
Done. This took some debugging to figure out. The 1st argument to `getStringObserver` must also be the name of the polymer property to sync with. I might change that in a PR because it caught me a little by surprise.

Reply:
Specifically, I think we can change `getStringObserver`'s signature to be

```typescript
getStringObserver(
    propertyName: string,
    defaultVal: string,
    useLocalStorage = false,
    polymerModuleProperty?: string,): Function
```

and `polymerModuleProperty` would only apply if `useLocalStorage` were false.